client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -63,7 +82,32 @@ private void init() {
         } catch (IOException e) {
             throw new RuntimeException("Error reading file " + privateKeyFile, e);
         }
-        var apiClient = ClientUtils.createApiClient(service, clientDid, privateKeyData);
-        this.registryApiClient = new RegistryApi(apiClient);
+
+        // temporary to preserve the backwards compatibility
+        if (dataspaceDid.isEmpty()) {
+            var apiClient = createApiClient(service, clientDid, privateKeyData);
+            this.registryApiClient = new RegistryApi(apiClient);
+            return;
+        }
+
+        registryApiClient = new RegistryApi(createApiClient(registrationUrl(), clientDid, privateKeyData));
+    }
+
+    private String registrationUrl() {
+        var didWebResolver = new WebDidResolver(httpClient(), useHttpsScheme(), MAPPER, new ConsoleMonitor());
+        var urlResolver = new RegistrationUrlResolver(didWebResolver);
+        return urlResolver.resolveUrl(dataspaceDid).orElseThrow(() -> new CliException("Error resolving the registration url."));
+    }
+
+    @NotNull
+    private OkHttpClient httpClient() {
+        return new OkHttpClient.Builder()
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .build();
+    }
+
+    private boolean useHttpsScheme() {
+        return parseBoolean(propOrEnv(USE_HTTPS_SCHEME, "true"));
```

Comment:
why should a CLI use here environment variable rather than an argument for this?

Reply:
to be consistent with EDC: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/728dd2dd109461bd799efe3ac9e4386655a808d6/extensions/iam/decentralized-identity/identity-did-web/src/main/java/org/eclipse/dataspaceconnector/iam/did/web/WebDidExtension.java#L43

Added comment to the env var.

Reply:
EDC is meant to be used in a server side app, where ENV vars are common, but I agree with @algattik that doing the same for CLI is very uncommon and a parameter would be more appropriate.

Reply:
Indeed, a CLI application does not take its configuration in the same manner as a server app